client/src/hooks/networks/utils.ts

```diff
@@ -0,0 +1,211 @@
+  type === 'organization'
+    ? String([
+        'country',
+        'lead_projects.country_of_coordination',
+        'lead_projects.lead_partner.country',
+        'lead_projects.partners.country',
+        'lead_projects.funders.country',
+        'partner_projects.country_of_coordination',
+        'partner_projects.lead_partner.country',
+        'partner_projects.partners.country',
+        'partner_projects.funders.country',
+        'funded_projects.country_of_coordination',
+        'funded_projects.lead_partner.country',
+        'funded_projects.partners.country',
+        'funded_projects.funders.country',
+      ])
+    : String([
+        'country_of_coordination',
+        'lead_partner.country',
+        'lead_partner.lead_projects.country_of_coordination',
+        'lead_partner.partner_projects.country_of_coordination',
+        'lead_partner.funded_projects.country_of_coordination',
+        'partners.country',
+        'partners.lead_project.country_of_coordination',
+        'partners.partner_projects.country_of_coordination',
+        'partners.funded_projects.country_of_coordination',
+        'funders.country',
+        'funders.lead_projects.country_of_coordination',
+        'funders.partner_projects.country_of_coordination',
+        'funders.funded_projects.country_of_coordination',
+      ]);
+
+// Get country data from the organization or project
+export const getCountryData = (
+  d: OrganizationListResponseDataItem | ProjectListResponseDataItem,
+  type: 'organization' | 'project',
+) =>
+  type === 'organization'
+    ? (d as OrganizationListResponseDataItem)?.attributes?.country?.data?.attributes
+    : (d as ProjectListResponseDataItem)?.attributes?.country_of_coordination?.data?.attributes;
+
+// Get only important data for the map
+export const getParsedData: (
+  d:
+    | OrganizationListResponseDataItem
+    | OrganizationLeadProjectsDataItem
+    | ProjectListResponseDataItem
+    | ProjectLeadPartnerData,
+  type: 'organization' | 'project',
+  countryD: OrganizationCountryDataAttributes | ProjectCountryOfCoordinationDataAttributes,
+) => ParsedData = (d, type, countryD) => ({
+  id: d.id,
+  type: type,
+  name: d.attributes?.name,
+  countryName: countryD?.name,
+  countryLat: countryD?.lat || 0,
+  countryLong: countryD?.long || 0,
+});
+
+// Parse data for each organization or project arrays
+export const parseData = (data: Data, type: 'organization' | 'project'): ParsedData[] => {
+  if (!data?.data) return [];
+  return (Array.isArray(data.data) ? data.data : [data.data])
+    ?.map((d) => {
+      const countryData = getCountryData(d, type);
+      if (!countryData) return null;
+      return getParsedData(d, type, countryData);
+    })
+    .filter((d): d is ParsedData => d !== null);
+};
+
+// Get the category of the organization or project for the chart
+const getCategory: (category: ProjectKey | OrganizationKey) => Category = (category) =>
```

Comment:
I think this can be simplified 😉:
```suggestion
const getCategory = (category: ProjectKey | OrganizationKey): Category =>
```